### ISSUE TITLE
Screen reader text generated for data points on charts table is incorrect for comparison periods

### ISSUE THREAD

**Issue by joelclimbsthings** [woocommerce/woocommerce-admin#6255]
**Describe the bug**

The charts utilized in the Analytics Reports pages generate screen reader text for each data point. These are conditionally displayed as part of the chart, and allow accessibility of this data to users who use a screen reader. 

Note that conditionally in this case refers to [the condition here](https://github.com/woocommerce/woocommerce-admin/blob/311fd0858f3b63717107aad8bc8ad589ffdc45ef/packages/components/src/chart/d3chart/utils/line-chart.js#L149), which only generates these points if there aren't too many, and the view-port is wide enough. 

It was [noticed in the context of another issue](https://github.com/woocommerce/woocommerce-admin/pull/6227#pullrequestreview-580876299) that the text generated incorrectly always uses the current period date, even for the points generated for the comparison period. Note that the correct data is shown in the tooltip, but not printed in the `aria-label` text on the data points.

The relevant line of code that generates these points[ is here](https://github.com/woocommerce/woocommerce-admin/blob/311fd0858f3b63717107aad8bc8ad589ffdc45ef/packages/components/src/chart/d3chart/utils/line-chart.js#L174).

**To Reproduce**

1. Enable a screen reader
2. Navigate to an Analytics Reports page, and select a period that has some, but not too many data points
3. Ensure a comparison period is selected to be displayed.
4. Navigate to the chart via keyboard, and use `tab` to focus the data points. 
5. The screen reader will reader out the date for the data point, with the incorrect date for the comparison period.

**Expected behavior**
The `aria-label` text for comparison data points should be correct

**Screenshots**

![image](https://user-images.githubusercontent.com/444632/106673587-aab52580-6566-11eb-9cb2-9546584b9fa9.png)
In the screenshot above, the selected point is for the comparison period so the `aria-label` should read _December 8, 2020_.

Instead it reflects the current period, as you can see in the source:

![image](https://user-images.githubusercontent.com/444632/106673783-05e71800-6567-11eb-8864-c1f3c027a4ff.png)
